tests/e2e/internal/framework/helm.go

```diff
@@ -0,0 +1,159 @@
+}
+
+// NewHelmClient creates a new helm client
+func NewHelmClient(opts ...HelmClientOption) (*HelmClient, error) {
+	client := &HelmClient{}
+	for _, o := range opts {
+		o(client)
+	}
+
+	var err error
+	client.repositoryCache, err = os.MkdirTemp("", ".helmcache")
+	if err != nil {
+		return nil, err
+	}
+
+	client.repositoryConfig, err = os.MkdirTemp("", ".helmrepo")
+	if err != nil {
+		return nil, err
+	}
+
+	restConfOptions := &helm.RestConfClientOptions{
+		Options: &helm.Options{
+			Namespace:        client.namespace,
+			RepositoryConfig: client.repositoryConfig,
+			RepositoryCache:  client.repositoryCache,
+		},
+		RestConfig: client.k8sRestConfig,
+	}
+
+	helmClient, err := helm.NewClientFromRestConf(restConfOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	client.client = helmClient
+
+	return client, nil
+}
+
+// HelmWithKubeConfig sets a kubeconfig value in the HelmClient struct
+func HelmWithKubeConfig(kubeconfig *restclient.Config) HelmClientOption {
+	return func(c *HelmClient) {
+		c.k8sRestConfig = kubeconfig
+	}
+}
+
+// HelmWithNamespace sets a namespace value in the HelmClient struct
+func HelmWithNamespace(namespace string) HelmClientOption {
+	return func(c *HelmClient) {
+		c.namespace = namespace
+	}
+}
+
+// HelmWithChart sets a chart value in the HelmClient struct
+func HelmWithChart(chart string) HelmClientOption {
+	return func(c *HelmClient) {
+		c.chart = chart
+	}
+}
+
+type HelmChartOptions struct {
+	CleanupOnFail bool
+	GenerateName  bool
+	ReleaseName   string
+	Timeout       time.Duration
+	Wait          bool
+	DryRun        bool
+}
+
+// Install deploys the helm chart
+func (c *HelmClient) Install(ctx context.Context, params []string, chartOpts HelmChartOptions) (string, error) {
+	values := helmValues.Options{
+		Values: params,
+	}
+
+	chartSpec := helm.ChartSpec{
+		ChartName:     c.chart,
+		Namespace:     c.namespace,
+		GenerateName:  chartOpts.GenerateName,
+		Wait:          chartOpts.Wait,
+		Timeout:       chartOpts.Timeout,
+		CleanupOnFail: chartOpts.CleanupOnFail,
+		DryRun:        chartOpts.DryRun,
+		ValuesOptions: values,
+	}
+
+	if !chartOpts.GenerateName {
+		if len(chartOpts.ReleaseName) == 0 {
+			return "", fmt.Errorf("release name must be provided the GenerateName chart option is unset")
+		}
+		chartSpec.ReleaseName = chartOpts.ReleaseName
+	}
+
+	res, err := c.client.InstallChart(ctx, &chartSpec, nil)
+
+	if err != nil {
+		return "", fmt.Errorf("error installing the chart: %w", err)
+	}
+
+	return res.Name, err
+}
+
+func (c *HelmClient) Uninstall(releaseName string) error {
+	return c.client.UninstallReleaseByName(releaseName)
+}
+
+func (c *HelmClient) Cleanup() error {
+	err := os.RemoveAll(c.repositoryCache)
+	if err != nil {
+		return fmt.Errorf("failed to delete directory: %s :%w", c.repositoryCache, err)
+	}
+
+	err = os.RemoveAll(c.repositoryConfig)
+	if err != nil {
+		return fmt.Errorf("failed to delete directory: %s :%w", c.repositoryConfig, err)
```

Comment:
Please use existing convenation for error messages:
```
fmt.Errorf("failed to delete directory %s; err: %w", c.repositoryConfig, err)
```
OR
```
fmt.Errorf("failed to delete directory '%s'; err: %w", c.repositoryConfig, err)
```

Reply:
done